test/e2e_node/memory_manager_test.go

```diff
@@ -706,7 +707,139 @@ var _ = SIGDescribe("Memory Manager", "[LinuxOnly]", framework.WithDisruptive(),
+			isMultiNUMASupported = ptr.To(isMultiNUMA())
+		}
+
+		if len(allNUMANodes) == 0 {
+			allNUMANodes = getAllNUMANodes()
+		}
+	})
+
+	// dynamically update the kubelet configuration
+	ginkgo.JustBeforeEach(func(ctx context.Context) {
+		if len(ctnParams) > 0 {
+			testPod = makeMemoryManagerPod(ctnParams[0].ctnName, initCtnParams, ctnParams)
+			testPod.Spec.Resources = &v1.ResourceRequirements{
+				Limits: v1.ResourceList{
+					v1.ResourceCPU:    resource.MustParse("100m"),
+					v1.ResourceMemory: resource.MustParse("128Mi"),
+				},
+				Requests: v1.ResourceList{
+					v1.ResourceCPU:    resource.MustParse("100m"),
+					v1.ResourceMemory: resource.MustParse("128Mi"),
+				},
+			}
+		}
+	})
+
+	ginkgo.JustAfterEach(func(ctx context.Context) {
+		// delete the test pod
+		if testPod != nil && testPod.Name != "" {
+			e2epod.NewPodClient(f).DeleteSync(ctx, testPod.Name, metav1.DeleteOptions{}, f.Timeouts.PodDelete)
+		}
+	})
+
+	ginkgo.Context("with static policy", func() {
+		tempSetCurrentKubeletConfig(f, func(ctx context.Context, initialConfig *kubeletconfig.KubeletConfiguration) {
+			kubeParams := *defaultKubeParams
+			kubeParams.policy = staticPolicy
+			updateKubeletConfigWithMemoryManagerParams(initialConfig, &kubeParams)
+			if initialConfig.FeatureGates == nil {
+				initialConfig.FeatureGates = make(map[string]bool)
+			}
+			initialConfig.FeatureGates["PodLevelResources"] = true
+		})
+
+		ginkgo.Context("", func() {
+			ginkgo.BeforeEach(func() {
+				// override pod parameters
+				ctnParams = []memoryManagerCtnAttributes{
+					{
+						ctnName: "memory-manager-none",
+						cpus:    "100m",
+						memory:  "128Mi",
+					},
+				}
+			})
+
+			ginkgo.JustAfterEach(func() {
+				// reset containers attributes
+				ctnParams = []memoryManagerCtnAttributes{}
+				initCtnParams = []memoryManagerCtnAttributes{}
+			})
+
+			ginkgo.It("should not report any memory data during request to pod resources List when pod has pod level resources", func(ctx context.Context) {
+				testPod = e2epod.NewPodClient(f).CreateSync(ctx, testPod)
+
+				endpoint, err := util.LocalEndpoint(defaultPodResourcesPath, podresources.Socket)
+				framework.ExpectNoError(err)
+
+				var resp *kubeletpodresourcesv1.ListPodResourcesResponse
+				gomega.Eventually(ctx, func(ctx context.Context) error {
+					cli, conn, err := podresources.GetV1Client(endpoint, defaultPodResourcesTimeout, defaultPodResourcesMaxSize)
+					if err != nil {
+						return err
+					}
+					defer conn.Close() //nolint:errcheck
+					resp, err = cli.List(ctx, &kubeletpodresourcesv1.ListPodResourcesRequest{})
+
+					return err
+				}, time.Minute, 5*time.Second).Should(gomega.Succeed())
+
+				for _, podResource := range resp.PodResources {
+					if podResource.Name != testPod.Name {
+						continue
+					}
+
+					for _, containerResource := range podResource.Containers {
+						gomega.Expect(containerResource.Memory).To(gomega.BeEmpty())
+					}
```

Comment:
if container level resources are set as I can see in ctnParams, the expected pod will have container resources. So this looks incorrect

Reply:
In addition to this, it seems the existing checks are failing because the code detects `pod.Spec.Resources == nil` despite the test setting it. Do we forgot to enable some validation or some other FG?